### PR TITLE
update(notify): add warning it's own type

### PR DESCRIPTION
**Pull request by mikigoalie**

### CODE REVIEW

**Review by LukeWasTakenn — COMMENTED** [review]
Is it me or does that exclamation svg not look super centered, at least compared to the other notification icons?

**Comment by mikigoalie** [comment]
Not really sure, I just fitted so it's same as other defaults, it looks good with circle though

**Comment by mikigoalie** [comment]
[Image preview](https://ctrlv.link/shots/2023/04/14/LTRd.png)
Idk, looks ok to me now. I'm just lazy to build it every time and I'd welcome default warning notification

File: web/src/features/notifications/NotificationWrapper.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ const Notifications: React.FC = () => {
         break;
     }
     if (!data.icon) {
-      data.icon = data.type === 'error' ? 'xmark' : data.type === 'success' ? 'check' : 'info';
+      data.icon = data.type === 'error' ? 'xmark' : data.type === 'success' ? 'check' : data.type === 'warning' ? 'circle-exclamation' : 'info';
```
Comment (thelindat) [inline web/src/features/notifications/NotificationWrapper.tsx:121]: Should probably swap this out for a switch.

**Comment by N-fire** [comment]
Get yourself appealed on VVarden Discord https://discord.gg/warden

**Comment by mikigoalie** [comment]
Done, clear

**Review by LukeWasTakenn — APPROVED** [review]

**Comment by mikigoalie** [comment]
@LukeWasTakenn @thelindat could you please check latest commit? I've had problems where if you spammed cancel it would sometimes incorrectly cancel prog bars, that seems to fix it

**Comment by mikigoalie** [comment]
The declaration and progress = nil was there already, it wasn't added by me, however, there is condition race issue if you try and spam starting and cancelling the progress bar. If you are quick enough, you can start prog bar before its cancelled

Repro. Bind progress to a key, spam the key and cancelling, if done quick enough, progress bar will result as true although it should be cancelled

**Comment by thelindat** [comment]
how many times do I have to say to keep PRs in a limited scope.

**Pull request closed without merging**